frontend/src/modules/app/ToolBar.tsx

```diff
@@ -163,74 +154,69 @@ export const ToolBar = ({
   const alarmLimitsRequestDraftSelect = useSelector(getAlarmLimitsRequestDraft);
   const alarmLimitsRequestSelect = useSelector(getAlarmLimitsRequest);
   const alarmLimitsRequestUnsaved = useSelector(getAlarmLimitsRequestUnsaved);
-  const alarmLimitsUnsavedKeys = useSelector(getAlarmLimitsUnsavedKeys);
   const alarmLimitsRequest = (alarmLimitsRequestSelect as unknown) as Record<string, Range>;
   const alarmLimitsRequestDraft = (alarmLimitsRequestDraftSelect as unknown) as Record<
     string,
     Range
   >;
   const ventilatingStatus = useSelector(getVentilatingStatusChanging);
-  /**
-   * State to manage toggling ventilationState
-   */
-  const [isVentilatorOn, setIsVentilatorOn] = React.useState(ventilating);
-  /**
-   * State to manage label for Landing page
-   */
-  const [landingLabel, setLandingLabel] = useState('Loading...');
+  const firmwareConnected = useSelector(getFirmwareConnected);
+  const backendConnected = useSelector(getBackendConnected);
   /**
    * State to manage ventilation label
    * Label is Dynamic based on ventilation state
    */
   const [label, setLabel] = useState('Start Ventilation');
   /**
-   * State to toggle if Ventilating isDisabled
+   * State to open Modal Popup for Dashboard
    */
-  const [isDisabled, setIsDisabled] = useState(false);
   const [discardOpen, setDiscardOpen] = useState(false);
+  /**
+   * State to open Modal Popup for Start/Pause button
+   */
+  const [startDiscardOpen, setStartDiscardOpen] = useState(false);
+
   const setAlarmLimitsRequestDraft = (data: Partial<AlarmLimitsRequest>) => {
     dispatch(commitDraftRequest<AlarmLimitsRequest>(MessageType.AlarmLimitsRequest, data));
   };
-  const alarmConfig = alarmConfiguration(currentMode);
-  // const isDisabled = !isVentilatorOn && location.pathname !== QUICKSTART_ROUTE.path;
+  const dispatchParameterRequest = (update: Partial<ParametersRequest>) => {
+    dispatch(commitRequest<ParametersRequest>(MessageType.ParametersRequest, update));
+  };
 
   /**
    * Updates Ventilation status on clicking Start/Pause ventilation
    */
   const updateVentilationStatus = () => {
-    if (!staticStart) {
-      dispatch(
-        commitRequest<ParametersRequest>(MessageType.ParametersRequest, {
-          ventilating: !isVentilatorOn,
-        }),
-      );
-      setIsVentilatorOn(!isVentilatorOn);
-    }
-    if (isVentilatorOn || staticStart) {
-      history.push(QUICKSTART_ROUTE.path);
+    if (ventilating) {
+      // if ventilating and there are unsaved alarm limit changes then open modal popup
+      if (alarmLimitsRequestUnsaved) {
+        setStartDiscardOpen(true);
+        return;
+      }
+    } else {
+      // if not ventilating and the firmware/backend is disconnected then return early
```

Comment:
Let's also add an explanation of why we should return early when the firmware/backend is disconnected.

Reply:
This probably will change considering the new scenarios that we discussed, although I'll add a comment of whatever i do in the end.